generator/README.md

```diff
@@ -127,6 +127,17 @@ modules:
                              #   EnumAsStateSet: An enum with a time series per state. Good for variable low-cardinality enums.
 ```
 
+### EnumAsInfo and EnumAsStateSet
+
+Sometimes it'll be necessary to specify an override for a metric because they are
+enum-like, i.e they provide a fixed number of possible options for a value.
+
+The difference between `EnumAsInfo` and `EnumAsStateSet` is that the first one should
```

Comment:
Let's split this into two paragraphs to avoid mixing information about each type. This will be less confusing for users to parse.